lib/rspec/mocks/verifying_proxy.rb

```diff
@@ -34,23 +57,25 @@ def add_message_expectation(method_name, opts={}, &block)
         ensure_implemented(method_name)
         super
       end
+    end
 
-      def ensure_implemented(method_name)
-        return unless method_reference[method_name].unimplemented?
+    # @private
+    module MaybeVerifyProxyMethods
```

Comment:
They were introduced to prevent having to *always* check, as I think the normal use case is not going to be to skip verification, given we check the permanent setting over in the space it seems prudent to do the same for this edge case.

The class hierarchy was introduced to avoid repetition, I could refactor that a bit more into module instead if you'd like? Or I could just do a bit of repetition :)